Dockerfile.ci

```diff
@@ -28,3 +23,9 @@ RUN wget -q -O /tmp/consul.zip https://releases.hashicorp.com/consul/${CONSUL_VE
     && sudo unzip -d /usr/local/bin /tmp/consul.zip \
     && rm -rf /tmp/consul*
 
+## Add the git repo and run bootstrap to install CI dependencies, then remove
+## the extra checkout to save image size.
+COPY GNUmakefile /tmp/build/GNUmakefile
+COPY scripts /tmp/build/scripts
+RUN cd /tmp/build && sudo chown -R $(whoami) /tmp/build && make deps lint-deps && rm -rf /tmp/build
```

Comment:
I moved this to bottom so we can cache effectively.

Also, once we copy the files and it's in the image, removing it in a subsequent layer doesn't reduce overall image size.  By importing only `GNUmakefile` and `scripts`, the image went down to 750MB from 1GB according to https://hub.docker.com/r/hashicorpnomad/ci-build-image/tags :
<img width="376" alt="Screen Shot 2019-08-23 at 1 48 53 PM" src="https://user-images.githubusercontent.com/136429/63612682-d4fe6780-c5ac-11e9-9b46-1279321d7d20.png">